File: Form1.cs
namespace Xichun_City_Train
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void 保存FToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Stream myStream;
            SaveFileDialog saveFileDialog1 = new SaveFileDialog();

            saveFileDialog1.Filter = "Xichun City Train Data files (*.xc-raod)|*.xc-road";
            saveFileDialog1.FilterIndex = 2;
            saveFileDialog1.RestoreDirectory = true;

            if (saveFileDialog1.ShowDialog() == DialogResult.OK)
            {
                if ((myStream = saveFileDialog1.OpenFile()) != null)
                {
                    // Code to write the stream goes here.
                    StreamWriter streamWriter = new StreamWriter(myStream);
                    streamWriter.WriteLine("车次：" + IDC_TEXTBOXTRAIN.Text);
                    streamWriter.WriteLine("车次类型：" + IDC_RICHTEXTBOXSTATION.Text);
                    streamWriter.WriteLine("机车：" + IDC_TEXTBOXLOCOMOTIVE.Text);
                    streamWriter.WriteLine("最大速度=" + IDC_TEXTBOXMAXSPEED.Text+"km/h");
                    streamWriter.WriteLine("所经站点：" +IDC_RICHTEXTBOXSTATION.Text);
                    streamWriter.Close();
                    myStream.Close();
                }
            }
        }
    }
}

File: Form1.vb
Imports System.IO

Public Class Form1
    Private Sub 保存ToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles 保存ToolStripMenuItem.Click
        Dim myStream As Stream
        Dim saveFileDialog1 As New SaveFileDialog()

        saveFileDialog1.Filter = "Huichun City Data files (*.hcdll)|*.hcdll"
        saveFileDialog1.FilterIndex = 2
        saveFileDialog1.RestoreDirectory = True

        If saveFileDialog1.ShowDialog() = DialogResult.OK Then
            myStream = saveFileDialog1.OpenFile()
            If (myStream IsNot Nothing) Then
                ' Code to write the stream goes here.
                Dim streamwriter As StreamWriter = New StreamWriter(myStream)
                streamwriter.WriteLine(RichTextBoxData.Text)
                streamwriter.Close()
                myStream.Close()
            End If
        End If
    End Sub

    Private Sub 打开ToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles 打开ToolStripMenuItem.Click
        Dim myStream As Stream = Nothing
        Dim openFileDialog1 As New OpenFileDialog()

        openFileDialog1.InitialDirectory = "c:\"
        openFileDialog1.Filter = "Huichun City Data files  (*.hcdll)|*.hcdll"
        openFileDialog1.FilterIndex = 2
        openFileDialog1.RestoreDirectory = True

        If openFileDialog1.ShowDialog() = System.Windows.Forms.DialogResult.OK Then
            Try
                myStream = openFileDialog1.OpenFile()
                If (myStream IsNot Nothing) Then
                    ' Insert code to read the stream here.
                End If
            Catch Ex As Exception
                MessageBox.Show("Cannot read file from disk. Original error: " & Ex.Message)
            Finally
                ' Check this again, since we need to make sure we didn't throw an exception on open.
                If (myStream IsNot Nothing) Then
                    myStream.Close()
                End If
            End Try
        End If
    End Sub
End Class
